stocking/src/main/scala/dev/atedeg/mdm/stocking/Actions.scala

```diff
@@ -20,18 +21,23 @@ def getMissingCountFromProductStock(
     availableStock: AvailableStock,
     desiredStock: DesiredStock,
 )(product: Product): MissingQuantity =
-  if availableStock(product).n >= desiredStock(product).n then MissingQuantity(0)
-  else MissingQuantity(desiredStock(product).n.toNonNegative - availableStock(product).n)
+  if availableStock.availableStock(product).n >= desiredStock.desiredStock(product).n then MissingQuantity(0)
+  else MissingQuantity(desiredStock.desiredStock(product).n.toNonNegative - availableStock.availableStock(product).n)
 
 /**
- * Removes the given quantity of a certain [[Product product]] from the [[AvailableStock stock]], giving the new current [[AvailableStock stock]].
+ * Removes the given quantity of a certain [[Product product]] from the [[AvailableStock stock]], giving the
+ * new current [[AvailableStock stock]].
  */
 def removeFromStock[M[_]: Monad: CanRaise[NotEnoughStock]](
     stock: AvailableStock,
 )(product: Product, quantity: Quantity): M[AvailableStock] =
-  (stock(product).n > quantity.n)
-    .otherwiseRaise(NotEnoughStock(product, quantity, stock(product)): NotEnoughStock)
-    .thenReturn(stock + (product -> AvailableQuantity(stock(product).n - quantity.n)))
+  (stock.availableStock(product).n > quantity.n)
+    .otherwiseRaise(NotEnoughStock(product, quantity, stock.availableStock(product)): NotEnoughStock)
+    .thenReturn(
+      AvailableStock(
+        stock.availableStock + (product -> AvailableQuantity(stock.availableStock(product).n - quantity.n)),
+      ),
+    )
```

Comment:
This could look better with a for comprehension
```suggestion
  for
    _ <- (stock.availableStock(product).n > quantity.n)
      .otherwiseRaise(NotEnoughStock(product, quantity, stock.availableStock(product)): NotEnoughStock)
    updatedStock = stock.availableStock.updatedWith(product)(_.map(q => AvailableQuantity(q.n - quantity.n)))
  yield AvailableStock(updatedStock)
```